### PR TITLE
Refactored Code To OOP-School_Library

### PR DESCRIPTION
**Pull request by AKeeganDev**
This pull request includes the following features 🤖:

📌 Added `book_generator.rb`, menu.rb`, persons_generator.rb`, `rental_generator.rb`, and `rentals_displayer.rb` helper classes.
📌 Fixed prior Rubocop errors (method length too long).
📌 Checked Ruby Style Guide to model the code structure.
📌 Checked code for KIS(s), DRY, YAGNI violations 🚫.
📌 Checked code to ensure proper use of SOLID 🧊.

### CODE REVIEW

File: main [hunk 1]
```diff
@@ -1,53 +1,27 @@
 #!/usr/bin/env ruby
 require_relative 'app'
+require_relative 'menu'
```
Comment (sja-thedude) [inline main]: - [ ] Kindly rename the file to main.rb, in hurry I think your missed to name file with .rb extension. Kindly do that in order to run the app smoothly.

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @AKeeganDev @Alexr16, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ Professional readme
✔️ Good PR title and description
✔️ All linters are passing

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main (hunk 1 above)
Comment (Whoistolu) [reply main]: Well resolved. Good refactor and  App works fine.

**Review by Whoistolu — APPROVED** [review]
### Hi Team
@AKeeganDev and @Alexr16 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**